src/renderer/components/+workspaces/add-workspace.tsx

```diff
@@ -0,0 +1,57 @@
+import React from "react";
+import { observer } from "mobx-react";
+import { Workspace, workspaceStore } from "../../../common/workspace-store";
+import { v4 as uuid } from "uuid";
+import { commandRegistry } from "../../../extensions/registries/command-registry";
+import { Input, InputValidator } from "../input";
+import { navigate } from "../../navigation";
+import { CommandOverlay } from "../command-palette/command-container";
+
+const uniqueWorkspaceName: InputValidator = {
+  condition: ({ required }) => required,
+  message: () => `Workspace with this name already exists`,
+  validate: value => !workspaceStore.workspacesList.find((workspace) => workspace.name === value),
+};
+
+@observer
+export class AddWorkspace extends React.Component {
+  onSubmit(name: string) {
+    if (!name.trim()) {
+      return;
+    }
+    const workspace = workspaceStore.addWorkspace(new Workspace({
+      id: uuid(),
+      name
+    }));
+
+    workspaceStore.setActive(workspace.id);
```

Comment:
Need to check for `undefined` value of `workspace` variable since `addWorkspace` method currently can return `undefined`  which is also not good. So I suggest to refactor `addWorkspace` or put some check before setting active workspace.

The reason is an error occured when trying to create workspace with already occupied name (eg `default`)
![Screenshot 2021-02-02 at 10 17 40](https://user-images.githubusercontent.com/9607060/106565466-e7b9f180-653f-11eb-93a8-fe95a6fd18b4.png)

Reply:
Hmm.. onSubmit should not happen if the value is not valid. Will try to reproduce.

Reply:
Reproduced, for some reason input still triggers `onSubmit` even if validation fails. Added more protection to add workspace logic.